src/Entities/EnderCrystal.cpp

```diff
@@ -21,9 +23,49 @@ cEnderCrystal::cEnderCrystal(Vector3d a_Pos, bool a_ShowBottom):
 
 
 
+void cEnderCrystal::SetShowBottom(bool a_ShowBottom)
+{
+	m_ShowBottom = a_ShowBottom;
+	if (m_World != nullptr)
+	{
+		m_World->BroadcastEntityMetadata(*this);
+	}
+}
+
+
+
+
+
+void cEnderCrystal::SetBeamTarget(Vector3i a_BeamTarget)
+{
+	m_BeamTarget = a_BeamTarget;
+	if (m_World != nullptr)
+	{
+		m_World->BroadcastEntityMetadata(*this);
+	}
+}
+
+
+
+
+
+void cEnderCrystal::SetDisplayBeam(bool a_DisplayBeam)
+{
+	m_DisplayBeam = a_DisplayBeam;
+	if (m_World != nullptr)
+	{
+		m_World->BroadcastEntityMetadata(*this);
+	}
+}
+
+
+
+
+
 void cEnderCrystal::SpawnOn(cClientHandle & a_ClientHandle)
 {
 	a_ClientHandle.SendSpawnEntity(*this);
+	a_ClientHandle.SendEntityMetadata(*this);
```

Comment:
not sure, does SendSpawnEntity do this already?

Reply:
It didn't display anything I changed until i added this line.
I checked: this doesn't happen